caddyhttp/httpserver/server.go

```diff
@@ -126,15 +126,16 @@ func (s *Server) Listen() (net.Listener, error) {
 		ln = tcpKeepAliveListener{TCPListener: tcpLn}
 	}
 
+	cln := ln.(caddy.Listener)
```

Comment:
It's needed to cast the net.Listener into a caddy.Listener for use in the middleware immediately loop below it.